site/content/how-to/traffic-management/response-headers.md

```diff
@@ -0,0 +1,199 @@
+
+
+```yaml
+kubectl apply -f - <<EOF
+apiVersion: gateway.networking.k8s.io/v1
+kind: HTTPRoute
+metadata:
+  name: headers
+spec:
+  parentRefs:
+  - name: gateway
+    sectionName: http
+  hostnames:
+  - "cafe.example.com"
+  rules:
+  - matches:
+    - path:
+        type: PathPrefix
+        value: /headers
+    backendRefs:
+    - name: headers
+      port: 80
+EOF
+```
+
+### Send Traffic to the Basic Headers Application
+
+To access the application, we will use `curl` to send requests to the `headers` endpoint. Notice our configured header values can be seen in the `responseHeaders` section below. We have four custom response headers defined - `X-Header-Unmodified`, `X-Header-Add`, `X-Header-Set`, `X-Header-Remove`.
+
+```shell
+curl -i --resolve cafe.example.com:$GW_PORT:$GW_IP http://cafe.example.com:$GW_PORT/headers
+```
+
+```text
+HTTP/1.1 200 OK
+Server: nginx/1.25.5
+Date: Mon, 06 May 2024 19:08:39 GMT
+Content-Type: text/plain
+Content-Length: 2
+Connection: keep-alive
+X-Header-Unmodified: unmodified
+X-Header-Add: add-to
+X-Header-Set: overwrite
+X-Header-Remove: remove
+
+ok
+```
+
+### Configure the HTTPRoute with Response Header Modifiers
+
+Let's configure the HTTPRoute with `ResponseHeaderModifier` filter:
+
+```yaml
+kubectl apply -f - <<EOF
+apiVersion: gateway.networking.k8s.io/v1
+kind: HTTPRoute
+metadata:
+  name: headers
+spec:
+  parentRefs:
+  - name: gateway
+    sectionName: http
+  hostnames:
+  - "cafe.example.com"
+  rules:
+  - matches:
+    - path:
+        type: PathPrefix
+        value: /headers
+    filters:
+    - type: ResponseHeaderModifier
+      responseHeaderModifier:
+        set:
+        - name: X-Header-Set
+          value: overwritten-value
+        add:
+        - name: X-Header-Add
+          value: this-is-the-appended-value
+        remove:
+        - X-Header-Remove
+    backendRefs:
+    - name: headers
+      port: 80
+EOF
+```
+
+This HTTPRoute has a few important properties:
+
+- The `parentRefs` references the gateway resource that we created, and specifically defines the `http` listener to attach to, via the `sectionName` field.
+- `cafe.example.com` is the hostname that is matched for all requests to the backends defined in this HTTPRoute.
+- The `match` rule defines that all requests with the path prefix `/headers` are sent to the `headers` Service.
+- There is a `ResponseHeaderModifier` filter defined for the path prefix `/headers`. This filter sets the value for the header `X-Header-Set` to `overwritten-value`, adds the value `this-is-the-appended-value` to the header `X-Header-Add`, and removes `X-Header-Remove` header.
+
+
+### Send Traffic to the Modified Headers Application
+
+Notice our configured header values can be seen in the `responseHeaders` section below, and that the `X-Header-Remove` header is absent. The header `X-Header-Add` gets appended with the new value and `X-Header-Set` gets overwritten to `overwritten-value` as defined in the *HttpRoute*.
```

Comment:
Let's move this below the output from the curl command. Let's also add a sentence describing what we are doing. Basically, we want to send another curl request to the`/headers` and verify that the response headers are modified.